LibreNMS/Util/RrdTool.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace LibreNMS\Util;
+
+use App\Models\Device;
+use LibreNMS\Config;
+use LibreNMS\Data\Store\Rrd;
+
+
+class RrdTool
+{
+    /**
+     * Get array of all rrd files for device
+     *
+     * @param array $device device for which we get the rrd's
+     * @return array $rrdfilearray array of rrd files for this host
+     */
+    public static function getRrdFiles($device)
+    {
+        if (Config::get('rrdcached')) {
+            $rrdcachedcmd = sprintf('%s list /%s --daemon %s', Config::get('rrdtool', 'rrdtool'), $device['hostname'], Config::get('rrdcached'));

```

Comment:
You should send rrdtool commands to the listening instance rather than spawn a new process.
(Unless I am miss-remembering and the process is not always available)

Reply:
yeah it looks like it would make sense to put it in LibreNMS/Data/Store/Rrd.php so i'll rework this.